packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -917,4 +920,1034 @@ moduleFor('Components test: curly components', class extends RenderingTest {
     this.assertElement(this.firstChild.childNodes[2], { tagName: 'b', content: 'bold' });
   }
 
+  ['@test lookup of component takes priority over property']() {
+    this.registerComponent('some-component', {
+      template: 'some-component'
+    });
+
+    this.render(
+      '{{some-prop}} {{some-component}}',
+      {
+        'some-component': 'not-some-component',
+        'some-prop': 'some-prop'
+      }
+    );
+
+    this.assertText('some-prop some-component');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('some-prop some-component');
+  }
+
+  ['@test component without dash is not looked up']() {
+    this.registerComponent('somecomponent', {
+      template: 'somecomponent'
+    });
+
+    this.render(
+      '{{somecomponent}}',
+      {
+        'somecomponent': 'notsomecomponent'
+      }
+    );
+
+    this.assertText('notsomecomponent');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('notsomecomponent');
+
+    this.runTask(() => this.context.set('somecomponent', 'not not notsomecomponent'));
+
+    this.assertText('not not notsomecomponent');
+
+    this.runTask(() => this.context.set('somecomponent', 'notsomecomponent'));
+
+    this.assertText('notsomecomponent');
+  }
+
+  ['@test non-block with properties on attrs']() {
+    this.registerComponent('non-block', {
+      template: 'In layout - someProp: {{attrs.someProp}}'
+    });
+
+    this.render('{{non-block someProp=prop}}', {
+      prop: 'something here'
+    });
+
+    this.assertText('In layout - someProp: something here');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('In layout - someProp: something here');
+
+    this.runTask(() => this.context.set('prop', 'other thing there'));
+
+    this.assertText('In layout - someProp: other thing there');
+
+    this.runTask(() => this.context.set('prop', 'something here'));
+
+    this.assertText('In layout - someProp: something here');
+  }
+
+  ['@skip non block with properties overridden in init']() {
```

Comment:
@chancancode I added this test. My apologies on removing it. It is skipped because in every kind of `rerender` the value set on invokation wins.